### DISCUSSION TITLE
How to configure c/ c ++ compilation environment and debug the project?

### DISCUSSION THREAD

**Discussion by mfkwgizhoujian** [eclipse-theia/theia#8145] [General] [answered]
### Steps to Reproduce:

I want use theia to run C/C++ code，but i do not know to to configure compilation environment and debug.
I installed the cpptools.vsix, it can't work correctly, but the same configuration in my vscode can work normally.
I read the theia/packages/cpp/README, but I don't understand it at all.
is there any example? thanks.

### Additional Information

- Operating System: MacOS Catalina 10.15.5
- Theia Version: latest

**Comment by vince-fugnitto** [comment, marked as answer]
@mfkwgizhoujian thank you for the question, in the future please open questions under 'discussions' as per what is mentioned in the issue templates.

We have not tested the functionality or compatibility of the `cpptools` plugin since it's [license](https://marketplace.visualstudio.com/items/ms-vscode.cpptools/license) is restricted to only be used by official vscode products. You may take a look at our [C/C++ docker image](https://github.com/theia-ide/theia-apps/tree/master/theia-cpp-docker) which setups the environment and includes the necessary dependencies and vscode extensions for development.
